ports/espressif/common-hal/busio/UART.c

```diff
@@ -46,12 +48,20 @@ void uart_reset(void) {
             continue;
         }
         #endif
-        if (uart_is_driver_installed(num)) {
+        if (uart_is_driver_installed(num) && !(never_reset_uart_mask & 1 << num)) {
             uart_driver_delete(num);
         }
     }
 }
 
+void common_hal_busio_uart_never_reset(busio_uart_obj_t *self) {
+    common_hal_never_reset_pin(self->rx_pin);
+    common_hal_never_reset_pin(self->tx_pin);
+    common_hal_never_reset_pin(self->rts_pin);
+    common_hal_never_reset_pin(self->cts_pin);
+    never_reset_uart_mask |= 1 << self->uart_num;
```

Comment:
Should `common_hal_busio_uart_deinit` reset the mask?

Reply:
Nope as `never_reset` is meant for things that should be kept alive across multiple `VM` and `REPL` runs. 
These things should only be reset on reboot of the mcu.

Reply:
Sounds good, I thought I had seen it reset the never_reset for displayio if you release the display. The rest looks good to me.

Reply:
Generally deinit should undo never_reset. Never reset is only meant to protect against bulk reset. Deiniting a specific instance should still work.

Reply:
That makes sense... `never_reset` should protect from things like `reset_port` and not from an explicit `deinit` call.

At a lot of places I find that the `never_reset` mask isn't being reset on a `deinit` call, this shouldn't create any issues with the functionality of things but should be fixed in the long term. I'll open an issue for this.